pkg/convert/pprof/profile.go

```diff
@@ -187,35 +191,41 @@ func (p *RawProfile) Parse(ctx context.Context, putter storage.Putter, _ storage
 		if p.SampleTypeConfig != nil {
 			sampleTypes = p.SampleTypeConfig
 		}
-		p.parser = NewParser(ParserConfig{
-			SpyName:             md.SpyName,
-			Labels:              md.Key.Labels(),
-			Putter:              putter,
-			SampleTypes:         sampleTypes,
-			SkipExemplars:       p.SkipExemplars,
-			StackFrameFormatter: StackFrameFormatterForSpyName(md.SpyName),
-		})
 
-		if p.PreviousProfile != nil {
-			// Ignore non-cumulative samples from the PreviousProfile
-			// to avoid duplicates: although, presence of PreviousProfile
-			// tells that there are cumulative sample types, it may also
-			// include regular ones.
-			filter := p.parser.sampleTypesFilter
-			p.parser.sampleTypesFilter = func(s string) bool {
-				if filter != nil {
-					return filter(s) && sampleTypes[s].Cumulative
-				}
-				return sampleTypes[s].Cumulative
+		if p.StreamingParser {
+			parser := streaming.VTStreamingParserFromPool(streaming.ParserConfig{
+				SpyName:       md.SpyName,
+				Labels:        md.Key.Labels(),
+				Putter:        putter,
+				SampleTypes:   sampleTypes,
+				SkipExemplars: p.SkipExemplars,
+			})
+			p.parser = parser
+			if p.PoolStreamingParser {
+				defer func() {
+					parser.ReturnToPool()
+					p.parser = nil
+				}()
```

Comment:
Use of `streaming.VTStreamingParserFromPool` should depend on `p.PoolStreamingParser` as far as I understand

Reply:
it is, it is in a nested if branch.
or i don't get the comment?

Reply:
I mean that a parser should be only taken from pool if `p.PoolStreamingParser` is true, else it should be created with `NewStreamingParser`. Otherwise we will pollute the pool because we never return to it, which may cause higher memory consumption and some extra pressure on GC. However, `sync.Pool` implementation is quite smart and the impact might be unnoticeable.

Also, it is a common practice (according to my observations) to explicitly call `Reset` before returning an item to the pool in order to release (and make them available to GC) objects that can't be reused.

Frankly, I don't think the current implementation is wrong or not optimal, but there are some things that you are used to and expect to see